docs/source/design/threaded_runner.rst

```diff
@@ -0,0 +1,68 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_threaded_runner:
+
+
+****************
+Threaded runner
+****************
+
+.. topic:: Specification scope and status
+
+   This specification provides an overview over the current implementation.
+
+Threads
+=======
+
+Datalad often requires the execution of subprocesses. While subprocesses are executed, datalad, i.e. its main thread, should be able to read data from stdout and stderr of the subprocess as well as write data to stdin of the subprocess. This requires a way to efficiently multiplex reading from stdout and stderr of the subprocess as well as writing to stdin of the subprocess.
+
+Since non-blocking IO and waiting on multiple sources (poll or select) differs wastly in terms of capabilities and API on different OSs, we decided to use blocking IO and threads to multiplex reading from different sources.
```

Comment:
```suggestion
Since non-blocking IO and waiting on multiple sources (poll or select) differs vastly in terms of capabilities and API on different OSs, we decided to use blocking IO and threads to multiplex reading from different sources.
```